### DISCUSSION TITLE
Existing Bundles Issue  after Migration from 6.9.6 to 10x

### DISCUSSION THREAD

**Discussion by ARY112** [pimcore/pimcore#12178] [Q&A / Support]
Till Pimcore 6.9.6, Bundles were getting registered via composer update and respective bundle services were working.

Composer.json
  "autoload": {
    "psr-4": {
      "": ["src/"],
	  
Bundles   
Folder Structure for Bundles and all file names as per Dependency Injection:
src==>AdvancedObjectSearchBundle with NameSpace AdvancedObjectSearchBundle
src==>DropdownLoadBundle with NameSpace DropdownLoadBundle
src==>ImageAssetsBundle with NameSpace ImageAssetsBundle

Extension.php Example
"AdvancedObjectSearchBundle\\AdvancedObjectSearchBundle" => TRUE,
"DropdownLoadBundle\\DropdownLoadBundle" => TRUE,
"ImageAssetsBundle\\ImageAssetsBundle" => TRUE,

For Pimcore 10x, Bundles are not getting registered.

Composer.json ()
  "autoload": {
    "psr-4": {
      "App\": ["src/"],  This is mandatory as per documentaion.
	  

We tried to add App in extension before Bundle Name and same inside before the namespace of all classes of Bundle otherwise composer gives error then Bundles got registered and enabled.Folder structure is same. It might be breaking name convention needed by Dependency Injection.
	  
Extension.php after changes
"App\\AdvancedObjectSearchBundle\\AdvancedObjectSearchBundle" => TRUE,
"App\\DropdownLoadBundle\\DropdownLoadBundle" => TRUE,
"App\\ImageAssetsBundle\\ImageAssetsBundle" => TRUE,	  

Example for Namespace after changes
namespace App\AdvancedObjectSearchBundle;

We are facing issue in Services now.

Not able to use the services from bundle in pimcore 10.Getting below error message

Service "app.product_header_service" not found: even though it exists in the app's container, the container inside "App\VendorCustomImportBundle\Controller\VendorImportController" is a smaller service locator that only knows about the "Pimcore\Http\Request\Resolver\DocumentResolver", "Pimcore\Http\Request\Resolver\EditmodeResolver", "Pimcore\Http\Request\Resolver\ResponseHeaderResolver", "Pimcore\Templating\Renderer\EditableRenderer", "doctrine", "form.factory", "http_kernel", "message_bus", "messenger.default_bus", "parameter_bag", "pimcore.templating", "request_stack", "router", "security.authorization_checker", "security.csrf.token_manager", "security.token_storage", "serializer", "session" and "twig" services. Try using dependency injection instead.
Server threw exception - could not perform action. Please reload the admin interface and try again.

Entry in Services.Yaml 

app.product_header_service:
class: 'App\VendorCustomImportBundle\Services\ProductHeader'
tags: ['container.ProductHeader']
arguments:
- '@Pimcore\Bundle\AdminBundle\Translation\AdminUserTranslator'
- '@product_helper'
- '@spreadsheet_helper'
- '@custom_user'
public: true

Kindly suggest what are we missing here. Is it the right way to do it.

**Comment by dvesh3** [comment]
@ARY112 Hey, either use FQCN service name and inject it using DI or include the service using `getSubscribedServices()` to access it from container locator e.g. https://github.com/pimcore/pimcore/blob/10.x/bundles/AdminBundle/Controller/AdminController.php#L44

**Comment by Thejaswini-Rao-U** [comment]
@ARY112 ,
im getting the issue, for registering existing bundles in pimcore10 version

can you please suggest how you followed?

i added in composer.json file => "SAMLBundle\": "src/SAMLBundle",,
bundles.php file => SAMLBundle\SAMLBundle::class => ['all' => true],
then config.yaml file => 
pimcore:
bundles:
search_paths:
- SAMLBundle

also added in kernel.php file as

if (class_exists('\SAMLBundle\SAMLBundle')) {
$collection->addBundle(new \SAMLBundle\SAMLBundle);
}

is this correct way?

after adding these all, i cant see this bundle when running  bin/console assets:install

Also migration were successfull, but in the admin interface i get this error.

![router](https://github.com/pimcore/pimcore/assets/78895325/09efa537-5710-4fc9-89f7-de12cccb7826)

Can anyone help me to solve this issue, tried many things but no luck

**Comment by Thejaswini-Rao-U** [comment]
Its fixed

**Reply by elburro1887** [reply]
How did you fix this? I had the same issue, but then set the following in `config/config.yaml`:
```
web_profiler:
  toolbar: false
  intercept_redirects: false
```

This fixed the issue, but the web profiler is now completely disabled. is there way to fix this without disabling it? 
Im on `10.6.9`

**Reply by Thejaswini-Rao-U** [reply]
@elburro1887 , You need to install symfony profiler , it will fix the issue
